### DISCUSSION TITLE
Introduce Product Collection Block Layout Options

### DISCUSSION THREAD

**Discussion by shaunandrews** [woocommerce/woocommerce#45052] [WooCommerce, Blocks, and Site Editing]
Our upcoming Product Collection block allows for builders to switch between two types of layouts: grid and list. Right now this option is toggled using buttons in the block's toolbar:

<img width="331" alt="The block toolbar showing icon buttons for grid and list views." src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/3710a7eb-44a0-421f-8cb8-3a354fb507fa">

--

In the video below, I show a new "Layout" selector in the Collection block's inspector. Three options are shown: Stacked, Grid, and Scroll. Stack and Grid map to our existing views. I'm not 100% with changing to Stack, but it does line up with WP Core Group block's Stack option.

The new Scroll option forces the list of products to extend out of the viewport, using horizontal scrolling/dragging (along with next/prev buttons) to browse. This is a common pattern I've noticed as I look around some popular e-commerce stores, especially for product recommendations, and landing pages.

https://github.com/woocommerce/woocommerce-blocks/assets/191598/8739d2fa-e604-4606-9ac7-15578ab7256c

<img width="1568" alt="CleanShot 2023-08-22 at 15 56 21@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/191598/4c231806-bbda-4804-99f5-5f5f38ff80aa">

**Comment by kmanijak** [comment]
> The new Scroll option forces the list of products to extend out of the viewport, using horizontal scrolling/dragging (along with next/prev buttons) to browse. This is a common pattern I've noticed as I look around some popular e-commerce stores, especially for product recommendations, and landing pages.

Do you have some specific expectations about how changing pages should look (or some example that showcases the behavior you have in mind) or should we suggest something? I mean stuff like animation, timing, and carousel behavior as a whole.
